node/network/availability-recovery/src/lib.rs

```diff
@@ -972,7 +1007,9 @@ impl AvailabilityRecoverySubsystem {
 				}
 				output = state.ongoing_recoveries.select_next_some() => {
 					if let Some((candidate_hash, result)) = output {
-						state.availability_lru.put(candidate_hash, result);
+						if let Ok(result) = CachedValidity::try_from(result) {
```

Comment:
This is a bit of a mind-twister, we take a result, and get a `Ok(result)`, this should be `Ok(cachable_validity)`

Reply:
haha,  I haven't noticed - that was really bad.